### PR TITLE
adopt fp endpoint

### PR DESCRIPTION
**Pull request by jeremy-babylonlabs**
updated the fp states and adopt v2 fp endpoints

### CODE REVIEW

File: src/app/api/getFinalityProviders.ts [hunk 1]
```diff
@@ -64,7 +67,7 @@ export const getFinalityProviders = async ({
 
   const response = await apiWrapper(
     "GET",
-    "/v1/finality-providers",
+    `/v2/finality-providers`,
```
Comment (gbarkhatov) [inline src/app/api/getFinalityProviders.ts:70]: Nitpick: I think we don't need the backticks here `, and can use " instead

**Review by gbarkhatov — APPROVED** [review]

File: src/app/types/finalityProviders.ts [hunk 2]
```diff
@@ -16,3 +16,18 @@ export interface Description {
...
+  FINALITY_PROVIDER_STATUS_ACTIVE: "Active",
+  FINALITY_PROVIDER_STATUS_JAILED: "Jailed",
+  FINALITY_PROVIDER_STATUS_SLASHED: "Slashed",
+};
+
+export const mapFinalityProviderState = (
+  state: FinalityProviderState,
+): string => {
+  return stateMap[state];
+};
```
Comment (totraev) [inline src/app/types/finalityProviders.ts:33]: @jeremy-babylonlabs `type` files should not contain instances or implementations. Move this logic to `FinalityProvider` component
```suggestion
export type FinalityProviderState = "FINALITY_PROVIDER_STATUS_INACTIVE" | "FINALITY_PROVIDER_STATUS_ACTIVE " | "  FINALITY_PROVIDER_STATUS_JAILED" | "FINALITY_PROVIDER_STATUS_SLASHED" ;
```

File: src/app/components/Staking/FinalityProviders/FinalityProvider.tsx [hunk 3]
```diff
@@ -23,7 +27,7 @@ interface FinalityProviderProps {
```
Comment (totraev) [inline src/app/components/Staking/FinalityProviders/FinalityProvider.tsx:27]:
```suggestion
const stateMap = {
  FINALITY_PROVIDER_STATUS_INACTIVE: "Inactive",
  FINALITY_PROVIDER_STATUS_ACTIVE: "Active",
  FINALITY_PROVIDER_STATUS_JAILED: "Jailed",
  FINALITY_PROVIDER_STATUS_SLASHED: "Slashed",
} as const;
```

File: src/app/components/Staking/FinalityProviders/FinalityProvider.tsx [hunk 4]
```diff
@@ -130,7 +134,8 @@ export const FinalityProvider: React.FC<FinalityProviderProps> = ({
           />
         </div>
         <div className="flex justify-start gap-1 capitalize lg:justify-end">
-          <span className="lg:hidden">Status:</span> {state}
+          <span className="lg:hidden">Status:</span>{" "}
+          {mapFinalityProviderState(state as FinalityProviderState)}
```
Comment (totraev) [inline src/app/components/Staking/FinalityProviders/FinalityProvider.tsx:138]:
```suggestion
          {stateMap[state]}
```

**Review by totraev — APPROVED** [review]

**Pull request merged**